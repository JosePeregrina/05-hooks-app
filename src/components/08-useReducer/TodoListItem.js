import React from 'react';
import PropTypes from 'prop-types';

function TodoListItem({ todo, index, handleDelete, handleToggle }) {
  const { id, desc, done } = todo;
  return (
    <li className='list-group-item'>
      <p className={`${done && 'complete'}`} onClick={() => handleToggle(id)}>
        {' '}
        {index + 1} - {desc}{' '}
      </p>
      <button className='btn btn-danger ms-3' onClick={() => handleDelete(id)}>
        Borrar
      </button>
    </li>
  );
}

export default TodoListItem;

TodoListItem.propTypes = {
  todo: PropTypes.object.isRequired,
  index: PropTypes.number.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleToggle: PropTypes.func.isRequired,
};
