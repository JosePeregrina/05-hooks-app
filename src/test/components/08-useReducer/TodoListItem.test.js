import { shallow } from 'enzyme';
import TodoListItem from '../../../components/08-useReducer/TodoListItem';
import { demoTodos } from '../../fixures/demoTodos';

describe('<TodoListItem />', () => {
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();

  let wrapper = shallow(
    <TodoListItem
      todo={demoTodos[0]}
      index={0}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );

  beforeEach(() => {
    wrapper = shallow(
      <TodoListItem
        todo={demoTodos[0]}
        index={0}
        handleDelete={handleDelete}
        handleToggle={handleToggle}
      />
    );
  });

  test('debe de mostrarse correctamente ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de llamar a la función borrar con el id del todo', () => {
    wrapper.find('button').simulate('click', demoTodos[0].id);

    expect(handleDelete).toHaveBeenCalled();
    expect(handleDelete).toHaveBeenCalledWith(1);
  });

  test('debe de llamar a la función toogle con el id del todo', () => {
    wrapper.find('p').simulate('click', demoTodos[0].id);

    expect(handleToggle).toHaveBeenCalled();
    expect(handleToggle).toHaveBeenCalledWith(1);
  });

  test('debe de mostrar este texto correctamente ', () => {
      const message = wrapper.find('p').text().trim() 
      expect(message.includes(demoTodos[0].desc)).toBe(true);
  })

  test('debe de tener la clase complete si el todo esta completado', () => {
    const wrapper = shallow(
        <TodoListItem
          todo={demoTodos[1]}
          index={1}
          handleDelete={handleDelete}
          handleToggle={handleToggle}
        />
      );

      expect(wrapper.find('p').hasClass('complete')).toBeTruthy()
  })
  
  
});
