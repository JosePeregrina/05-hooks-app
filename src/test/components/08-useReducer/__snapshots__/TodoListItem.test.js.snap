// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoListItem /> debe de mostrarse correctamente  1`] = `
<li
  className="list-group-item"
>
  <p
    className="false"
    onClick={[Function]}
  >
     
    1
     - 
    Aprender React
     
  </p>
  <button
    className="btn btn-danger ms-3"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
